libs/server-sdk/src/data_systems/lazy_load/lazy_load_system.cpp

```diff
@@ -20,15 +28,17 @@ void LazyLoad::Shutdown() {}
 
 std::shared_ptr<data_model::FlagDescriptor> LazyLoad::GetFlag(
     std::string const& key) const {
-    auto const state = tracker_.State(Keys::kAllSegments, time_());
+    auto const state =
+        tracker_.State(data_components::DataKind::kFlag, key, time_());
```

Comment:
todo: I thought this was a typo (using `kAllSegments` here) - am I using it wrong?